### ISSUE TITLE
[Feature request] Specification inheritance

### ISSUE THREAD

**Issue by rrozendo** [likec4/likec4#1349]
First of all, thanks (and congratulations) for this awesome modelling tool

I am considering using LikeC4 as the official modelling tool in my company. One thing that I found really useful is the ability to define default style options for custom-defined elements, such as in:

```
specification {

  element container

  element database {
    style {
      shape storage
    }
  }

  element queue {
    style {
      shape queue
    }
  }
}
```

It's a nice way to standardize multiple diagrams across the company. However, one could argue that it would also be nice to explicitly define that both "database" and "queue" are types of containers to stick to the C4 standard abstractions. Just like the concept of inheritance in object-oriented programming, we would define in this example that every database is a container but not every container is a database, and every queue is a container but not every container is a queue. Maybe it could be something like this:

```
specification {

  element container {

    element database {
        style {
          shape storage
        }
    }

     element queue {
      style {
        shape queue
      }
    }

  }
}
```

This way it could maintain flexibility of defining custom elements while allowing architects to keep on using technically only the standard C4 levels of abstractions if they really want to

**Comment by a-scolan** [comment]
Agreed, would help maintain more structured and complex projects, especially with multi-project setups

**Comment by Kiiv** [comment]
Interested in this too, specifically to work around #1705 ;)

A keyword should be great
```
 specification {

  element parent {
        style {
          shape storage
        }
    }

  element child extend parent {
      style {
        color color_stockage
      }
    }

}
```

**Comment by baterflyrity** [comment]
Our team is using LikeC4 for big projects. We really need this feature to address 20x diagrams manual update for e.g. storage provider change.
